File: csharp/DiagramPrinter/DiagramPrintDriver.cs
namespace DiagramPrinter;

public class DiagramPrintDriver
{
    private FlowchartDiagram? _diagram;

    public void SetDiagram(FlowchartDiagram diagram)
    {
        this._diagram = diagram;
    }

    public void ReleaseDiagram()
    {
        this._diagram = null;
    }

    public virtual bool PrintTo(PhysicalPrinter physicalPrinter)
    {
        throw new NotImplementedException("Can't call this from a unit test");
    }
}

File: java/src/main/java/sammancoaching/DiagramPrintDriver.java
package sammancoaching;

public class DiagramPrintDriver {
    private FlowchartDiagram diagram;

    public void setDiagram(FlowchartDiagram diagram) {
        this.diagram = diagram;
    }

    public void releaseDiagram() {
        this.diagram = null;
    }

    public boolean printTo(PhysicalPrinter physicalPrinter) {
        throw new UnsupportedOperationException("Can't call this from a unit test");
    }
}
